credsweeper/config/config.py

```diff
@@ -33,5 +33,8 @@ def __init__(self, config: Dict) -> None:
         self.find_by_ext: bool = config["find_by_ext"]
         self.size_limit: Optional[int] = parse_size(config["size_limit"]) if config["size_limit"] is not None else None
         self.depth: int = int(config["depth"])
-        if self.depth > 0 and ".zip" in self.exclude_extensions:
-            self.exclude_extensions.remove(".zip")
+        if 0 < self.depth:
+            if ".zip" in self.exclude_extensions:
+                self.exclude_extensions.remove(".zip")
+            if ".gz" in self.exclude_extensions:
+                self.exclude_extensions.remove(".gz")
```

Comment:
@kmnls here

Reply:
The solution is inconsistent. If the user provides zip as a file type to skip you should not silently remove this exclude list.
I propose to check zip and gz only if they are not excluded manually.

Reply:
@csh519 You opinion?

Reply:
@kmnls 

As you told this solution is inconsistent.. It would be good to there is another way to extract files for zip files(data scan)..

I think `--depth` scanning option is option for CredSweeper, and it means the user agreed to scan zip files.
So it doesn't matter if `.zip`, `.gzip` extensions are in config file.
I propose to add new file extractor for zip file or other possible file types. (Maybe `FileForceExtractor`..?)

Reply:
The feature --depth was introduced after issue #116. It is only partially solution and use only standard libraries. 
Assume, --depth will help research APK files that are not excluded. Thus, ZIP extension is do. 
So, I did the workaround for zip files to prove the solution. BTW, xlsx, docx and so on are zip archives too.
Therefore we have to exclude pure binary files e.g. bmp, jpg and so on. Quickest way is skip by extension.

Proposal 1 (easiest):
- add exclude list for binary files which are (might be) containers (zip, gzip, tar, iso ...) 
- when --depth is not defined - exclude found file from both exclusion lists. 
- when --depth is defined - exclude only from standard list
Proposal 2 :
- add totally file recognition by signature and skip unsupported binary files (e.g. git supposes that file is binary when 0x00 is in first 8000bytes, with some workaround for utf-16...) + something like https://github.com/h2non/filetype.py.
- skip files by extension list as currently if --depth is not enabled;
- read all files and skip binary files which are not supported when --depth enabled

Let the PR will be merged as is and new feature will be created after solution designed.

Reply:
Before new issue creation - let decide what way preferred.

Reply:
I propose the following solution.
In the config file we preserve the excluded_extensions section.
To the config file we add our  "depth" option as 2 options "check_inside_archives"+"depth", plus we add additional list "include" (if "zip" format is in then it will be checked).
Thus, if the file is in excludes list then it will be skipped anyway, and if not then:
- if check_inside_archives + in includes then it will be unzipped and checked
- if not chec_inside_archives then it will be scanned as binary.